### PR TITLE
Fix: Remove misleading error logs

### LINKED ISSUE #1221: Logging 'Stack peek was null when {method}' is misleading for some Hub calls.
#1127 introduced changes on how to deal with the Scope stack but logging is off in a few cases.

eg:
`setFingerprint`
`setUser`
`clearBreadcrumbs`

If the stack can't be null, should we just remove them?

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
## :scroll: Description

Fix: Remove misleading error logs

## :bulb: Motivation and Context

Fixes #1221

## :pencil: Checklist

- [x] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [x] No breaking changes

### CODE REVIEW

File: sentry/src/main/java/io/sentry/Hub.java [hunk 1]
```diff
@@ -482,13 +479,8 @@ public void bindClient(final @NotNull ISentryClient client) {
           .log(SentryLevel.WARNING, "Instance is disabled and this 'bindClient' call is a no-op.");
     } else {
       final StackItem item = stack.peek();
-      if (client != null) {
-        options.getLogger().log(SentryLevel.DEBUG, "New client bound to scope.");
-        item.setClient(client);
-      } else {
-        options.getLogger().log(SentryLevel.DEBUG, "NoOp client bound to scope.");
-        item.setClient(NoOpSentryClient.getInstance());
-      }
```
Comment (marandaneto) [inline sentry/src/main/java/io/sentry/Hub.java:491]: I guess we should keep this, `client` can be null, it's not really about the stack peek being null like the others

Comment (maciejwalkowiak) [reply sentry/src/main/java/io/sentry/Hub.java:491]: `client` is marked as `@NotNull` in `StackItem`, is this incorrect?

Comment (marandaneto) [reply sentry/src/main/java/io/sentry/Hub.java:491]: `client` is a method param.

that's why we set either the `client` instance or `NoOpSentryClient.getInstance()`
you're correct about `StackItem` having non-null `client`

Comment (maciejwalkowiak) [reply sentry/src/main/java/io/sentry/Hub.java]: @marandaneto thanks, you're right. Fixed.

**Review by marandaneto — APPROVED** [review]

**Pull request merged**